test/with_dummyserver/test_connectionpool.py

```diff
@@ -229,20 +230,20 @@ def test_wrong_specific_method(self):
             r = pool.request("POST", "/specific_method", fields={"method": "GET"})
             assert r.status == 400, r.data
 
-    def test_upload(self):
+    def test_upload(self) -> None:
         data = "I'm in ur multipart form-data, hazing a cheezburgr"
-        fields = {
```

Comment:
Looking at `_TYPE_FIELDS` it seems like it should allow this, why did this need to change?

Reply:
Yeah, it seems so. but without these change we get:
```
test/with_dummyserver/test_connectionpool.py:243: error: Argument "fields" to "request" of "RequestMethods" has incompatible type "Dict[str, Sequence[str]]"; expected "Union[Sequence[Union[Tuple[str, Union[Union[str, bytes], Tuple[str, Union[str, bytes]], Tuple[str, Union[str, bytes], str]]], RequestField]], Dict[str, Union[Union[str, bytes], Tuple[str, Union[str, bytes]], Tuple[str, Union[str, bytes], str]]], None]"  [arg-type]
```

Do you have any idea what is the reason?

Reply:
Agreed here, it's really unfortunate that `str` can be inferred as `Sequence[str]`

Reply:
Thanks, @bluetech, I've added explicit annotation for `fields`.